### PR TITLE
[Magiclysm] Forge of wonders

### PR DESCRIPTION
**Pull request by Maleclypse**
clean well lit place
forge_southwest
forge_vault
forge_big

#### Summary

SUMMARY: Mods "New location and trader NPC"

#### Purpose of change
Adding a Magiclysm NPC who trades high end magical items and creating a believable location for this merchant to live out of. (Looking at you Zelda cave merchants)

#### Describe the solution

Create location special, mobs and npcs to populate it.  Furniture for the site, materials for the furniture, corpses for the mechanicals. I think that's it.  Forge demons and fish are spawning in good numbers.

#### Describe alternatives you've considered

none

#### Testing
See pictures, loads no errors.

### CODE REVIEW

File: data/mods/Magiclysm/itemgroups/itemgroups.json [hunk 1]
```diff
@@ -1132,5 +1132,42 @@
...
+    "subtype": "distribution",
+    "items": [
+      { "group": "enchanted_combat_items", "prob": 40 },
+      { "group": "enchanted_misc", "prob": 50 },
+      { "group": "enchanted_rings_unusual", "prob": 20 },
+      { "group": "enchanted_rings_rare", "prob": 30 },
+      { "group": "enchanted_wands_greater", "prob": 30 },
+      { "group": "enchanted_wands_lesser", "prob": 20 },
+      { "group": "lair_black_dragon_treasure", "prob": 5 },
+      { "group": "enchanted_melee_weapons_plus2" "prob": 20 }
```
Comment (LaVeyanFiend) [inline data/mods/Magiclysm/itemgroups/itemgroups.json:1170]:
```suggestion
      { "group": "enchanted_melee_weapons_plus2", "prob": 20 }
```

File: data/mods/Magiclysm/monsters/forgedwellers.json [hunk 2]
```diff
@@ -0,0 +1,145 @@
+[
+  {
+    "id": "mon_forgeborn",
+    "type": "MONSTER",
+    "name": { "str": "forge born" },
+    "description": "The most elaborate, gaudy and overwrought clockwork creature that you've ever seen.  Each one has it's face designed in a mask displaying a single emotion.  Blades extend from the wrists in combat.",
```
Comment (LaVeyanFiend) [inline data/mods/Magiclysm/monsters/forgedwellers.json:6]:
```suggestion
    "description": "The most elaborate, gaudy and overwrought clockwork creature that you've ever seen.  Each one has its face designed in a mask displaying a single emotion.  Blades extend from the wrists in combat.",
```

File: data/mods/Magiclysm/monsters/forgedwellers.json [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      "ANIMAL",
+      "PATH_AVOID_DANGER_1",
+      "NO_BREATHE"
+    ]
+  },
+  {
+    "id": "mon_forgedwarves",
+    "type": "MONSTER",
+    "name": { "str": "forge workers" },
+    "description": "A squat humanoid that looks quite sturdily built.  They style their hair in mohawks and other strange fashions that seem to be dictated by what parts of their skin aren't scarred from burns.  Their clothes are designed to protect them from the intense environment they work in.",
```
Comment (LaVeyanFiend) [inline data/mods/Magiclysm/monsters/forgedwellers.json:116]: If they're wearing protective equipment for the forge, you could add some armor_fire values. Applies to the rest of the forge faction.

**Comment by Maleclypse** [comment]
@curstwist is there something I'm doing wrong with the npc `"shopkeeper_item_group": "NC_FORGE_LORD_STORE",`? Shopkeeper doesn't trade any of the items in that group until I think he pulls it out of his bag of holding then equips it. Then he'll trade it.

**Comment by Maleclypse** [comment]
Assuming nothing that is broken that was missed in testing, this PR is ready for final review.

File: data/mods/Magiclysm/items/enchanted.json [hunk 4]
```diff
@@ -72,5 +72,23 @@
         }
       ]
     }
+  },
+  {
+    "id": "cloak_morthylla",
+    "type": "ARMOR",
+    "copy-from": "cloak_wool",
+    "name": { "str": "Cloak of Morthylla the Lamia" },
+    "description": "This cloak seems to absorb light except for the pinpricks of stars that move across it's surface.",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/items/enchanted.json:81]:
```suggestion
    "description": "This cloak seems to absorb light except for the pinpricks of stars that move across its surface.",
```

File: data/mods/Magiclysm/npc/TALK_FORGE_MERCHANT.json [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    "id": "TALK_FORGE_LORD",
+    "type": "talk_topic",
+    "dynamic_line": {
+      "u_is_wearing": "badge_marshal",
+      "yes": [
+        "The government doesn't subsidize your purchases here anymore.",
+        "Do you have any tales of adventure I could write down?",
+        "In case you were wondering, we no longer respect any pre-existing weapon laws."
+      ],
+      "no": [ "Welcome to the Forge of Wonders.", "Here to trade, I hope?", "Bring me, tales and we can trade." ]
```
Comment (KorGgenT) [inline data/mods/Magiclysm/npc/TALK_FORGE_MERCHANT.json:76]:
```suggestion
      "no": [ "Welcome to the Forge of Wonders.", "Here to trade, I hope?", "Bring me tales and we can trade." ]
```

File: data/mods/Magiclysm/npc/TALK_FORGE_MERCHANT.json [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+      { "text": "What are you doing here?", "topic": "TALK_FORGE_LORD_DOING" },
+      { "text": "Well, bye.", "topic": "TALK_DONE" }
+    ]
+  },
+  {
+    "id": "TALK_FORGE_LORD_DOING",
+    "type": "talk_topic",
+    "dynamic_line": {
+      "u_is_wearing": "badge_marshal",
+      "yes": "Well here and now I am the law and the law here is people purchase my goods they take them out into the world and they die or they return to buy more of my goods, either way I write poetry about them.",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/npc/TALK_FORGE_MERCHANT.json:89]:
```suggestion
      "yes": "Well here and now I am the law, and the law here is that people purchase my goods, they take them out into the world and they die; or they return to buy more of my goods.  Either way, I write poetry about them.",
```

**Comment by KorGgenT** [comment]
this won't block this PR, it's more a reminder for myself

**Comment by KorGgenT** [comment]
image
is he supposed to sell his own cloak? it also doesn't appear as though his inventory includes the large collection of goods he's near

**Comment by Maleclypse** [comment]
> image
> is he supposed to sell his own cloak? it also doesn't appear as though his inventory includes the large collection of goods he's near

I'll add "NO_TRADE" to his cloak tonight. thank's for catching that.  The second part seems to be a larger trade issue I've been trying to understand, I can't seem to make people trade things that belong to them and are near them anymore.  Probably error on my part, but I wasn't sure if it was a more recent bug. Writing this I've realized I can test it at the FMC trader at Hub01 and see what they've done.

**Comment by Maleclypse** [comment]
Moved back to WIP until I understand why the eastern items are no longer faction owned.    Southern items remain faction owned.

**Comment by Maleclypse** [comment]
Thank's to curstwist for showing me what was holding this up.  I had to make the cloak max price, because `NO_TRADE` will actually make NPCs throw an item away.  I don't believe there is a flag currently that will prevent them from trading any item that they are willing to hold, wear or own.

**Comment by jbytheway** [comment]
I feel like giving something a huge price is also not a good solution to this.  When the player gets hold of it they will abuse this.

**Comment by Maleclypse** [comment]
> I feel like giving something a huge price is also not a good solution to this. When the player gets hold of it they will abuse this.
Is there an existing flag I could add to it to fix the issue or should I make the price expensive but in line with similar items?

**Comment by esotericist** [comment]
There isn't an existing flag for this, but i'm looking into maybe adding one. For the purposes of this PR, i think the item should be "reasonably" priced (as in, priced according to its actual functionality).

**Comment by esotericist** [comment]
Now that #42286 is merged, you should be able to use either of those flags on the cloak as appropriate. (i'm not super clear how possessive you want the person to be of it)

**Pull request merged**